libr/util/num.c

```diff
@@ -408,17 +409,54 @@ R_API ut64 r_num_get_input_value(RNum *num, const char *input_value) {
 	return value;
 }
 
+#define NIBBLE_TO_HEX(n) (((n) & 0xf) > 9 ? 'a' + ((n) & 0xf) - 10 : '0' + ((n) & 0xf))
 R_API char* r_num_as_string(RNum *___, ut64 n) {
-	char str[10];
+	char str[34]; // 8 byte * 4 chars in \x?? format
 	int stri, ret = 0;
 	int len = sizeof (ut64);
 	ut64 num = n;
 	str[stri=0] = 0;
 	while (len--) {
 		char ch = (num & 0xff);
-		if (ch>=33 && ch <127) {
+		if (ch>=32 && ch <127) {
 			str[stri++] = ch;
 			str[stri] = 0;
+#ifdef R_NUM_AS_STRING_ESCAPES
+		} else if (ch == 7) {
```

Comment:
looks spaguetti what about a switch table and a helper function to get only 1 char as argument and the string and returns the stri increment